pkg/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -17,6 +19,17 @@ PATH_HYPERKUBE_IMAGE_USED_FOR_LAST_COPY_KUBECTL="{{ .pathHyperKubeImageUsedForLa
 
 mkdir -p "{{ .pathKubeletDirectory }}" "$PATH_HYPERKUBE_DOWNLOADS"
 
+deleteOrphan() {
+  if [[ ! -f "$PATH_CLOUDCONFIG_FILES_OLD" ]]; then
+    touch "$PATH_CLOUDCONFIG_FILES_OLD"
+  fi
+  # sort for join to work
+  sort -o "$PATH_CLOUDCONFIG_FILES" "$PATH_CLOUDCONFIG_FILES"
+  # calculates old_files - new_files and deletes them
+  join -t $'\n' -v 1 "$PATH_CLOUDCONFIG_FILES_OLD" "$PATH_CLOUDCONFIG_FILES" | xargs -L1 rm -f
+  mv "$PATH_CLOUDCONFIG_FILES" "$PATH_CLOUDCONFIG_FILES_OLD"
+}
```

Comment:
Not sure about our general logging policy in this context, but would it not make sense to at least log what we delete? Unless we expect to delete huge number of files it might be helpful in case for whatever reason something got deleted that should not have.